lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1166,14 +1142,36 @@ def main():
         update_max_failure_ratio=dict(default=0, type='float'),
         update_order=dict(default=None, type='str'),
         user=dict(default='root'))
+
+    option_minimal_versions = dict(
+        dns=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        dns_options=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        dns_search=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        hostname=dict(docker_py_version='2.2.0', docker_api_version='1.25'),
+        tty=dict(docker_py_version='2.4.0', docker_api_version='1.25'),
+        secrets=dict(docker_py_version='2.1.0', docker_api_version='1.25'),
```

Comment:
I think `secrets` was only added in docker-py 2.3.0 (see docker/docker-py@d1038c422b2a069494476dd743cc06e11e8939e7).

Reply:
Hmm, yes by that commit it certainly looks like that but checking the changelog and documentation tells another story:

https://github.com/docker/docker-py/blob/master/docs/change-log.md#210

And compare the services documentation 2.0.0 vs 2.1.0:
https://docker-py.readthedocs.io/en/2.0.0/services.html
https://docker-py.readthedocs.io/en/2.1.0/services.html

Reply:
You're right. I also checked the source at tag `2.1.0`, and it contains that parameter. I guess they did some refactoring between 2.2.x and 2.3.0 which resulted in that commit...